File: Backend/tiendaVirtual/src/main/resources/static/js/registrar/objects/Empresa.js
export  class Empresa {

    constructor(nit, nombre, direccion) {
        this._nit = nit,
        this._nombre = nombre,
        this._direccion = direccion
    }

    get nit() {
        return this.nit;
    }

    get nombre() {
        return this.nombre;
    }

    get direccion() {
        return this.direccion;
    }

    set nit(nit) {
        this._nit = nit;
    }

    set nombre(nombre) {
        this._nombre = nombre;
    }

    set direccion(direccion) {
        this._direccion = direccion;
    }

}

File: Backend/tiendaVirtual/src/main/java/com/segavaDev/tiendaVirtual/repositories/entities/Empresa.java
package com.segavaDev.tiendaVirtual.repositories.entities;

import java.io.Serializable;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Data

@Entity
@Table(name = "Empresas")
public class Empresa implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Setter(AccessLevel.NONE)
    private long id;

    @Column(unique = true, nullable = false)
    @Setter(AccessLevel.NONE)
    private long nit;

    @Column(unique = true, nullable = false)
    private String nombre;

    @Column(unique = true, nullable = false)
    private String direccion;
    
}
